lib/ansible/module_utils/common/text/converters.py

```diff
@@ -8,8 +8,15 @@
 import codecs
 import json
 
+from ansible.module_utils.compat import typing as _t
 from ansible.module_utils._internal import _no_six
 
+_T = _t.TypeVar('_T')
+
+_NonStringPassthru: _t.TypeAlias = _t.Literal['passthru']
+_NonStringOther: _t.TypeAlias = _t.Literal['simplerepr', 'empty', 'strict']
+_NonStringAll: _t.TypeAlias = _t.Union[_NonStringPassthru, _NonStringOther]
```

Comment:
I believe this would look nicer:
```suggestion
_NonStringAll: _t.TypeAlias = _NonStringPassthru | _NonStringOther
```

Reply:
We can't use this yet as module_utils needs to still support python3.9, and causes:

```
ERROR: lib/ansible/module_utils/common/text/converters.py:18:31: operator: Unsupported left operand type for | ("<typing special form>")
ERROR: lib/ansible/module_utils/common/text/converters.py:18:31: valid-type: Invalid type alias: expression is not a valid type
```

We can however use `TypeAlias` because the `ansible.module_utils.compat.typing` uses `typing_extensions` for polyfill on older python versions.

Reply:
Ah, fair. Marking this as resolved.